### PR TITLE
OCPBUGS-34373: Bump library-go to fix routes/custom-host permission for externalCertificate

### PR DESCRIPTION
**Pull request by chiragkyal**
- Bumps https://github.com/openshift/library-go/commit/0064ad7 and fixes the routes/custom-host permission for externalCertificate
- Removes `ValidateHostExternalCertificate` from route admission

Fixes # https://issues.redhat.com/browse/OCPBUGS-34373

### CODE REVIEW

**Comment by chiragkyal** [comment]
Blocked by https://github.com/openshift/openshift-apiserver/pull/458

**Comment by chiragkyal** [comment]
/assign @Miciah 
for lgtm

**Comment by chiragkyal** [comment]
/assign @p0lyn0mial 
for approval

File: pkg/route/apiserver/admission/routehostassignment/adapter.go [hunk 1]
```diff
@@ -47,21 +47,3 @@ func ValidateHostUpdate(ctx context.Context, route, oldRoute *routeinternal.Rout
 
 	return hostassignment.ValidateHostUpdate(ctx, &external, &oldExternal, sarClient, opts)
 }
-
-func ValidateHostExternalCertificate(ctx context.Context, route, oldRoute *routeinternal.Route, sarClient routecommon.SubjectAccessReviewCreator, opts routecommon.RouteValidationOptions) field.ErrorList {
```
Comment (p0lyn0mial) [inline pkg/route/apiserver/admission/routehostassignment/adapter.go:51]: may I know why we are removing the admission?

Comment (chiragkyal) [reply pkg/route/apiserver/admission/routehostassignment/adapter.go:51]: We have removed the `ValidateHostExternalCertificate` function from library-go https://github.com/openshift/library-go/pull/1754

- EP: https://github.com/openshift/enhancements/pull/1652

To summarize: To make the `custom-host` permission requirement consistent between `spec.tls.externalCertificate` and `spec.tls.certificate` for route.

**Comment by lihongan** [comment]
/label qe-approved
The pre-merged test passed
```
$ oc get clusterversion
NAME      VERSION                                                AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.18.0-0.test-2024-11-14-100343-ci-ln-zxfcfjb-latest   True        False         55m     Cluster version is 4.18.0-0.test-2024-11-14-100343-ci-ln-zxfcfjb-latest

$ oc whoami 
testuser-47

$ oc patch route myedge -p '{"spec":{"tls":{"externalCertificate":{"name":"mytls"}}}}'
route.route.openshift.io/myedge patched

// change to another secret
$ oc patch route myedge -p '{"spec":{"tls":{"externalCertificate":{"name":"newtls"}}}}'
route.route.openshift.io/myedge patched

$ oc patch route myreen -p '{"spec":{"tls":{"externalCertificate":{"name":"newtls"}}}}'
route.route.openshift.io/myreen patched

$ oc patch route myedge -p '{"spec":{"tls":{"externalCertificate":null}}}'
route.route.openshift.io/myedge patched
$ oc get route myedge -oyaml | yq .spec.tls
termination: edge

$ oc patch route myedge -p '{"spec":{"tls":{"externalCertificate":{"name":"mytls"}}}}'
route.route.openshift.io/myedge patched
$ oc get route myedge -oyaml | yq .spec.tls
externalCertificate:
  name: mytls
termination: edge
```

File: pkg/route/apiserver/admission/routehostassignment/adapter.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/route/apiserver/admission/routehostassignment/adapter.go:51]: ah, okay. On some platforms, the Route API is served as a CRD, and the validation logic is common and defined in library-go.

**Pull request merged**